synthesizer/src/vm/finalize.rs

```diff
@@ -1258,6 +1258,22 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
 
                     // Insert the next committee into storage.
                     store.committee_store().insert(state.block_height(), next_committee)?;
+
+                    #[cfg(feature = "history")]
```

Comment:
In your observations, what happened next? How did the node recover and move forward?
The reason I ask is because if it recovers, then a correct block will be written and the history will be updated accordingly.

Performance-wise, this is the best place to store the data since you already have the `bonded` and `delegated` mappings in memory. Moving this later would require pulling them from DB.

Reply:
> if it recovers, then a correct block will be written and the history will be updated accordingly.

Ah that's great, yes that solves the issue